### PR TITLE
Navigation Bar and AppNavigator class

### PR DESCRIPTION
**Pull request by RishanR**
The AppNavigator class will be the class that is injected into all ViewModels and used to navigate between screens from the ViewModel. This PR shows an example of that in the HomeViewModel.

This PR also creates a NavigationBar component that will live for the whole lifetime of the app. However, you can hide or show the navigation bar on certain screens by adding screens to the `notShownScreens` list in `BottomNavigatoinBar`. Any screens in this list will not have the NavigationBar showing on them.

Demo:

<video src="https://github.com/adshayanB/ECE452/assets/36520027/c9bc636c-5551-4cf6-95bd-f887c7cb553d"/>

### CODE REVIEW

**Review by adshayanB — APPROVED** [review]
Looks great! Will integrate the nav bar for sign in

File: app/src/main/java/com/example/farmeraid/uicomponents/ButtonView.kt [hunk 1]
```diff
@@ -29,11 +31,12 @@ fun ButtonView(
             disabledContainerColor = DisabledPrimaryColour,
             disabledContentColor = DisabledWhiteContentColour,
         ),
-        enabled = !buttonUiState.isLoading || buttonUiState.enabled,
+        enabled = !buttonUiState.isLoading && buttonUiState.enabled,
```
Comment (adshayanB) [inline app/src/main/java/com/example/farmeraid/uicomponents/ButtonView.kt:34]: Nice catch

Comment (RishanR) [reply app/src/main/java/com/example/farmeraid/uicomponents/ButtonView.kt:34]: ty :)

**Review by SathursanS — APPROVED** [review]
LGTM

File: app/src/main/java/com/example/farmeraid/sign_in/views/SignInScreenView.kt [hunk 2]
```diff
@@ -42,74 +40,61 @@ import com.example.farmeraid.uicomponents.models.UiComponentModel
...
-                modifier = Modifier
-                    .fillMaxWidth()
-                    .height(50.dp)
-            )
+            Spacer(modifier = Modifier.height(20.dp))
+            Box(modifier = Modifier.padding(40.dp, 0.dp, 40.dp, 0.dp)) {
+                ButtonView(
+                    buttonUiState = state.buttonUiState,
+                    buttonUiEvent = UiComponentModel.ButtonUiEvent(
+                        onClick = {viewModel.login(username.value.toString(), password.value.toString()) }),
```
Comment (adshayanB) [inline app/src/main/java/com/example/farmeraid/sign_in/views/SignInScreenView.kt:91]: Nit: Spacing on last param

**Review by adshayanB — APPROVED** [review]
🚀

**Comment by adshayanB** [comment]
The demo looks really clean! Super sick to see the sign in page has the full flow!! Great work

**Review by SoumilDharaskar — APPROVED** [review]

**Pull request merged**